docs-src/design-decisions.md

```diff
@@ -0,0 +1,42 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Better modeling for implementations with multiple GatewayClass resources
+  (including some potential multi-tenant scenarios using Cluster API).
+* Easier to provide simple and secure defaults (only allow Gateways and Routes
+  in the same namespace by default).
+* GatewayClass parameters can live in the same namespace.
+* More RBAC flexibility (can allow users to create GatewayClasses in select
+  namespaces only or restrict visibility of GatewayClasses in some namespaces).
+
+### Advantages of a Cluster-Scoped GatewayClass
+* Consistent with StorageClass and IngressClass resources.
+* Discussions with sig-storage showed that StorageClass had worked well as a
+  cluster-scoped resource and that they were continuing to follow that pattern
+  for new Class resources.
+* Follows general pattern that resources managed by infrastructure provider
+  and/or cluster administrators are cluster scoped.
+* Impossible for namespace level admins to grant RBAC permissions for
+  GatewayClass resources. (With a namespace-scoped resource, anyone that could
+  configure RBAC within a namespace could grant GatewayClass access).
+* Clear that GatewayClasses can be used by multiple namespaces.
+* Simple globally unique names.
+* Works well for majority of use cases - likely only a few total GatewayClasses
+  per cluster.
```

Comment:
Prior experience in the Ingress API (which effectively has a cluster scoped class) shows this works well.

Reply:
Neither of which account for management clusters (see my other comments). Cluster resources are anti-pattern IMO, and the `GatewayClass` isn't even a pure cluster resource as it defines relationships to namespaces as part of its spec.

Reply:
> Prior experience in the Ingress API (which effectively has a cluster scoped class) shows this works well.

Another thought, it may work well for Ingress, but maybe not other use cases that we're trying to present to this group and satisfy with the Service API? And that's okay, but what I am hearing is that y'all do not think our use is in-scope. If that's the decision, and it seems to be, that is fair and allows us to move on and think about how else to tackle the problem.

What I do not think is fair, is to argue that one prior experience should exclude consideration of other experiences for what is in and out of scope.

Thanks again for working with us on this!

Reply:
The service-apis WG was established in order to evolve the Ingress API, so I'd argue that experience there is relevant. That's orthogonal to whether the WG scope changes over time (and I don't think this document prevents that).

Reply:
Fair

Reply:
But I’d also argue that evolve doesn’t just mean “satisfy in a different way.” I was under the impression that the scope also means to satisfy non-Ingress use cases. And even from an Ingress POV I think it’s important to consider the use cases I am raising.